File: level2/위장.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;

int solution(vector<vector<string>> clothes) {
    map<string, int> m;
    map<string, int>::iterator iter;
    
    int size = clothes.size();
    for(int i = 0; i < size; i++){
        iter = m.find(clothes[i][1]);
        if(iter == m.end()){
            m.insert(make_pair(clothes[i][1],1));
        }
        else{
            m[clothes[i][1]]++;
        }
    }
    
    
    int answer = 1;
    for(iter = m.begin(); iter != m.end(); iter++){
        answer *= iter->second+1;
    }
    return answer-1;
}

File: level2/위장.py
def solution(clothes):
    answer = 0
    
    dic = {}
    for i in clothes:
        if i[1] in dic:
            dic[i[1]].append(i[0])
        else:
            dic[i[1]] = []
            dic[i[1]].append(i[0])
    
    len_list = []
    for key in dic:
        len_list.append(len(dic[key]))
    
    for i in range(len(len_list)):
        tmp = len_list[i]
        for j in range(i+1,len(len_list)):
            tmp *= len_list[j]+1
        answer += tmp
    
    
    return answer

if __name__ == '__main__':
    clothes = input()
    
    print(solution(clothes))
